### PR TITLE
Implement zrevrangebylex command

### PR DESCRIPTION
**Pull request by nioshield**

### CODE REVIEW

**Review by git-hulk — DISMISSED** [review]
LGTM, and we can add some comments like https://github.com/KvrocksLabs/kvrocks/blob/unstable/src/redis_zset.cc#L293 
to clarify why we need the SeekForPrev when using the reverse range.

File: tests/tcl/tests/unit/type/zset.tcl [hunk 1]
```diff
@@ -390,63 +390,63 @@ start_server {tags {"zset"}} {
...
-            # assert_equal 0 [r zlexcount zset + \[c]
-            # assert_equal 0 [r zlexcount zset \[c -]
-            # assert_equal 8 [r zlexcount zset \[bar +]
-            # assert_equal 5 [r zlexcount zset \[bar \[foo]
-            # assert_equal 4 [r zlexcount zset \[bar (foo]
-            # assert_equal 4 [r zlexcount zset (bar \[foo]
-            # assert_equal 3 [r zlexcount zset (bar (foo]
-            # assert_equal 5 [r zlexcount zset - (foo]
-            # assert_equal 1 [r zlexcount zset (maxstring +]
+        #     assert_equal 9 [r zlexcount zset - +]
```
Comment (ShooterIT) [inline tests/tcl/tests/unit/type/zset.tcl]: please revert this change, since we don't change anything

**Review by git-hulk — APPROVED** [review]

**Review by ShooterIT — APPROVED** [review]

**Comment by ShooterIT** [comment]
Thanks @nioshield merged
